Plugins/Onnx/src/OnnxRuntimeBase.cpp

```diff
@@ -69,28 +68,50 @@ std::vector<float> Acts::OnnxRuntimeBase::runONNXInference(
 // the function assumes that the model has 1 input node and 1 output node
 std::vector<std::vector<float>> Acts::OnnxRuntimeBase::runONNXInference(
     Acts::NetworkBatchInput& inputTensorValues) const {
+  return runONNXInferenceMultiOutput(inputTensorValues).front();
+}
+
+// Inference functions for single-input, multi-output models
+
+std::vector<std::vector<float>>
+Acts::OnnxRuntimeBase::runONNXInferenceMultiOutput(
+    std::vector<float>& inputTensorValues) const {
+  Acts::NetworkBatchInput vectorInput(1, inputTensorValues.size());
+  for (size_t i = 0; i < inputTensorValues.size(); i++) {
+    vectorInput(0, i) = inputTensorValues[i];
+  }
+  auto vectorOutput = runONNXInferenceMultiOutput(vectorInput);
+  return vectorOutput[0];
+}
+
+std::vector<std::vector<std::vector<float>>>
+Acts::OnnxRuntimeBase::runONNXInferenceMultiOutput(
+    NetworkBatchInput& inputTensorValues) const {
   int batchSize = inputTensorValues.rows();
   std::vector<int64_t> inputNodeDims = m_inputNodeDims;
-  std::vector<int64_t> outputNodeDims = m_outputNodeDims;
+  std::vector<std::vector<int64_t>> outputNodeDims = m_outputNodeDims;
 
   // The first dim node should correspond to the batch size
   // If it is -1, it is dynamic and should be set to the input size
   if (inputNodeDims[0] == -1) {
     inputNodeDims[0] = batchSize;
   }
-  if (outputNodeDims[0] == -1) {
-    outputNodeDims[0] = batchSize;
+
+  bool ondMatch = true;
```

Comment:
ond ? I think the acts coding convention is to a avoid abbreviation if possible (or make them explicit)